client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -22,9 +22,12 @@
    private final Sequence startingPosition;
    @Getter
    private final long groupRefreshTimeMillis;
+   @Getter
+   private final long automaticCheckpointIntervalMillis;   
 
    public static final class ReaderGroupConfigBuilder {
-       long groupRefreshTimeMillis = 3000;
+       private long groupRefreshTimeMillis = 3000;
+       private long automaticCheckpointIntervalMillis = -1;
```

Comment:
The logic should be the opposite here. By default, checkpoints are automatic because if I don't care about checkpoints, then why should I know about them at all, automatic or manual? If I want manual checkpoints, then I enable it.

I also don't like the idea of requesting the user to set a timeout value. It is hard to determine what that value should be. Could we do something like that is not time-based? For example, if we are able to determine that we need to rebalance based on the current heuristic, then we schedule one, otherwise, we can just keep going.

If we do something like this, then the only time we would need would be to pace rebalances.